Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,828 @@
+        @"NSApplication" : [appCustomClass stringValue];
+    }
+  
+  [self addStandardObjects: objects
+               classString: customClassString
+               connections: appCons];
+  
+  // Add it to the document
+  [objects detach];
+  [doc addChild: objects];
+}
+   
+- (NSArray *) findSubclassesOf: (Class)clz
+                    inDocument: (NSXMLDocument *)document
+{
+  NSArray *result = nil;
+  NSMutableArray *subclasses = [GSObjCAllSubclassesOfClass(clz) mutableCopy];
+
+  [subclasses insertObject: clz atIndex: 0];
+  FOR_IN(Class, cls, subclasses)
+    {
+      NSString *className = NSStringFromClass(cls);
+      NSString *classNameNoNamespace = [className substringFromIndex: 2];
+      NSString *xmlClassName = [NSString stringWithFormat: @"%@%@",
+                                         [[classNameNoNamespace substringToIndex: 1] lowercaseString],
+                                         [classNameNoNamespace substringFromIndex: 1]];
+      NSString *xpath = [NSString stringWithFormat: @"//%@",xmlClassName];
+      result = [document nodesForXPath: xpath error: NULL];
+
+      if ([result count] > 0)
+        {
+          break;
+        }
+    }
+  END_FOR_IN(subclasses);
+
+  return result;
+}
+
+- (NSString *) controllerIdWithDocument: (NSXMLDocument *)document
+{
+  NSString *controllerId = nil;
+  NSArray *windowControllers = [self findSubclassesOf: [NSWindowController class]
+                                           inDocument: document];
+  NSArray *viewControllers = [self findSubclassesOf: [NSViewController class]
+                                         inDocument: document];
+  NSArray *controllerPlaceholders = [document nodesForXPath: @"//controllerPlaceholder"
+                                                      error: NULL];
+  
+  if ([windowControllers count] > 0)
+    {
+      NSXMLElement *ce = [windowControllers objectAtIndex: 0];
+      NSXMLNode *attr = [ce attributeForName: @"id"];
+      controllerId = [attr stringValue];
+      
+      FOR_IN(NSXMLElement*, o, windowControllers)
+        {
+          NSXMLElement *objects = (NSXMLElement *)[o parent];
+          NSArray *windows = [o nodesForXPath: @"//window" error: NULL];
+          FOR_IN(NSXMLNode*, w, windows)
+            {
+              [w detach];
+              [objects addChild: w];
+            }
+          END_FOR_IN(windows);
+        }
+      END_FOR_IN(windowControllers);
+    }
+  
+  if ([viewControllers count] > 0)
+    {
+      NSXMLElement *ce = [viewControllers objectAtIndex: 0];
+      NSXMLNode *attr = [ce attributeForName: @"id"];
+      controllerId = [attr stringValue];
+    }
+  
+  if ([controllerPlaceholders count] > 0)
+    {
+      NSXMLElement *ce = [controllerPlaceholders objectAtIndex: 0];
+      NSXMLNode *attr = [ce attributeForName: @"id"];
+      controllerId = [attr stringValue];
+    }
+  
+  return controllerId;
+}
+
+- (void) processCustomElement: (NSXMLElement *)coel
```

Comment:
I was sure that I put in a comment to this method, but now I am no longer able to find it. Sometimes GitHub can play tricks on you. I appologize if I add this before.

The whole method could be replaced by just extracting the id of the first responder early on and putting that id into the processChild:... method using it instead of "-1". That way the actual unarchiving would resolve all the reference for us. As far as I am aware we never use the fact that the first responder has the id "-1" in our code.

Reply:
You could extract that id with just on xpath statement. Something like `//objects/customObject[@sceneMemberID ="firstResponder"]/@id`

Reply:
Looking into doing this.

Reply:
Removed.